nexus/db-queries/src/db/queries/region_allocation.rs

```diff
@@ -117,10 +189,56 @@ pub fn allocation_query(
 
     let seed = seed.to_le_bytes().to_vec();
 
-    let size_delta =
-        params.block_size * params.blocks_per_extent * params.extent_count;
+    // Ensure that the multiplication doesn't overflow.
+    let requested_size: u64 = params
+        .block_size
+        .checked_mul(params.blocks_per_extent)
+        .ok_or(AllocationQueryError::RegionSizeOverflow)?
+        .checked_mul(params.extent_count)
+        .ok_or(AllocationQueryError::RegionSizeOverflow)?;
+
+    if requested_size > MAX_REGION_SIZE_BYTES {
+        return Err(AllocationQueryError::RequestedRegionOverMaxSize {
+            request: requested_size,
+            maximum: MAX_REGION_SIZE_BYTES,
+        });
+    }
+
+    // After the above check, unconditionally cast from u64 to i64. The value is
+    // low enough that this shouldn't truncate.
+    let requested_size: i64 = requested_size.try_into().unwrap();
+
+    // The Crucible Agent's current reservation factor is 25%, so add that here.
+    // Check first that the requested region size is divisible by this. This
+    // should basically never fail because all block sizes are divisible by 4.
+    if requested_size % 4 != 0 {
```

Comment:
This is contingent on the value of `reservation_percent`, and the enum `RegionReservationPercent` only having a single value?

If we add another value to that enum, this code will happily still compile (incorrectly), right?

I know this is pedantic, but could we:

1. Move up the
```
    let reservation_percent =
        crate::db::model::RegionReservationPercent::TwentyFive;
```
from below

2. Run these "mod by four, set factor = 4" checks in a match arm, based on `reservation_percent`?


Then, if we add another option for `RegionReservationPercent`, the code will complain about a missing match arm, which would help identify this spot.